In [1]:
#1 read csv
#2 select remarks col
#3 url open
#4 collect link or scrape
#5 insert after remark
import numpy as np
import pandas as pd
from pandas import Series,DataFrame
f = pd.ExcelFile('TH_7-eleven 14-18 Dec.xls')
f

In [2]:
seven_file = f.parse('Sheet1')
seven_file.head()

creation_time             TH_time distance_source  passenger1_id  \
0 2016-12-14 06:24:09 2016-12-14 13:24:09     IPHONE2-POI        7066160   
1 2016-12-14 07:21:47 2016-12-14 14:21:47    ANDROID2-POI        2247921   
2 2016-12-14 07:31:22 2016-12-14 14:31:22    ANDROID2-POI         455541   
3 2016-12-14 07:40:07 2016-12-14 14:40:07    ANDROID2-POI        6862205   
4 2016-12-14 07:44:38 2016-12-14 14:44:38    ANDROID2-POI       12278436   

              passenger1_name  \
0  Wilaiwan Dhrammajariyapong   
1                         Kai   
2                         Tee   
3                   Poyroi Ka   
4                        Care   

                                             remarks      state  
0  กรุณาแสดงข้อมูลในลิงค์นี้ให้พนักงาน 7-Eleven. ...  COMPLETED  
1                                                NaN  COMPLETED  
2  กรุณาแสดงข้อมูลในลิงค์นี้ให้พนักงาน 7-Eleven. ...  COMPLETED  
3  กรุณาแสดงข้อมูลในลิงค์นี้ให้พนักงาน 7-Eleven. ...  COMPLETED  
4                                                NaN  COMPLETED

In [3]:
seven_file.remarks.count()

42

In [4]:
seven_file['TH_time'].head()

0   2016-12-14 13:24:09
1   2016-12-14 14:21:47
2   2016-12-14 14:31:22
3   2016-12-14 14:40:07
4   2016-12-14 14:44:38
Name: TH_time, dtype: datetime64[ns]

In [5]:
seven_file.remarks[2]

u'\u0e01\u0e23\u0e38\u0e13\u0e32\u0e41\u0e2a\u0e14\u0e07\u0e02\u0e49\u0e2d\u0e21\u0e39\u0e25\u0e43\u0e19\u0e25\u0e34\u0e07\u0e04\u0e4c\u0e19\u0e35\u0e49\u0e43\u0e2b\u0e49\u0e1e\u0e19\u0e31\u0e01\u0e07\u0e32\u0e19 7-Eleven. https://goo.gl/TNxAVM'

In [6]:
import urllib2

In [7]:
seven_file.remarks[2].split('7-Eleven. ')[1]

u'https://goo.gl/TNxAVM'

In [8]:
def readfile(n):
    f = urllib2.urlopen(seven_file.remarks[n].split('7-Eleven. ')[1])
    page = f.read()
    return page.split('<p>Order ID</p>\n      <h2>')[1].split('<')[0]

In [9]:
s =readfile(6)
print s

tHTEYDl7tS


if seven_file['remarks'].isnull:
    print 'null'
else:
    print 'not null'

In [10]:
seven_file.head()

creation_time             TH_time distance_source  passenger1_id  \
0 2016-12-14 06:24:09 2016-12-14 13:24:09     IPHONE2-POI        7066160   
1 2016-12-14 07:21:47 2016-12-14 14:21:47    ANDROID2-POI        2247921   
2 2016-12-14 07:31:22 2016-12-14 14:31:22    ANDROID2-POI         455541   
3 2016-12-14 07:40:07 2016-12-14 14:40:07    ANDROID2-POI        6862205   
4 2016-12-14 07:44:38 2016-12-14 14:44:38    ANDROID2-POI       12278436   

              passenger1_name  \
0  Wilaiwan Dhrammajariyapong   
1                         Kai   
2                         Tee   
3                   Poyroi Ka   
4                        Care   

                                             remarks      state  
0  กรุณาแสดงข้อมูลในลิงค์นี้ให้พนักงาน 7-Eleven. ...  COMPLETED  
1                                                NaN  COMPLETED  
2  กรุณาแสดงข้อมูลในลิงค์นี้ให้พนักงาน 7-Eleven. ...  COMPLETED  
3  กรุณาแสดงข้อมูลในลิงค์นี้ให้พนักงาน 7-Eleven. ...  COMPLETED  
4                                                NaN  COMPLETED

In [14]:
seven_file['remarks'].fillna(0,inplace=True)

seven_file['link'] = ['text' if isnull(even_file['remarks']) else s]

In [15]:
seven_file.head()

creation_time             TH_time distance_source  passenger1_id  \
0 2016-12-14 06:24:09 2016-12-14 13:24:09     IPHONE2-POI        7066160   
1 2016-12-14 07:21:47 2016-12-14 14:21:47    ANDROID2-POI        2247921   
2 2016-12-14 07:31:22 2016-12-14 14:31:22    ANDROID2-POI         455541   
3 2016-12-14 07:40:07 2016-12-14 14:40:07    ANDROID2-POI        6862205   
4 2016-12-14 07:44:38 2016-12-14 14:44:38    ANDROID2-POI       12278436   

              passenger1_name  \
0  Wilaiwan Dhrammajariyapong   
1                         Kai   
2                         Tee   
3                   Poyroi Ka   
4                        Care   

                                             remarks      state  
0  กรุณาแสดงข้อมูลในลิงค์นี้ให้พนักงาน 7-Eleven. ...  COMPLETED  
1                                                  0  COMPLETED  
2  กรุณาแสดงข้อมูลในลิงค์นี้ให้พนักงาน 7-Eleven. ...  COMPLETED  
3  กรุณาแสดงข้อมูลในลิงค์นี้ให้พนักงาน 7-Eleven. ...  COMPLETED  
4                                                  0  COMPLETED

In [18]:
count = 0 

In [24]:
fw = open('collect_data.txt','w')
count = -1
lists = []
for n in seven_file.remarks:
    count = count+1
    #print count
    try:
        s =readfile(count)
        #fw.write(str(count)+','+str(s)+'\n')
    except:
        s = 'error'
        #fw.write(str(count)+'\n')
    lists.append(s)
fw.close()

In [25]:
seven_file['link'] = lists

In [33]:
seven_file

creation_time             TH_time distance_source  passenger1_id  \
0  2016-12-14 06:24:09 2016-12-14 13:24:09     IPHONE2-POI        7066160   
1  2016-12-14 07:21:47 2016-12-14 14:21:47    ANDROID2-POI        2247921   
2  2016-12-14 07:31:22 2016-12-14 14:31:22    ANDROID2-POI         455541   
3  2016-12-14 07:40:07 2016-12-14 14:40:07    ANDROID2-POI        6862205   
4  2016-12-14 07:44:38 2016-12-14 14:44:38    ANDROID2-POI       12278436   
5  2016-12-14 08:18:50 2016-12-14 15:18:50     IPHONE2-POI       20273963   
6  2016-12-14 08:34:00 2016-12-14 15:34:00    ANDROID2-POI        2683605   
7  2016-12-14 08:59:59 2016-12-14 15:59:59    ANDROID2-POI        6862205   
8  2016-12-14 09:12:22 2016-12-14 16:12:22    ANDROID2-POI        5941955   
9  2016-12-14 09:14:56 2016-12-14 16:14:56    ANDROID2-POI         306456   
10 2016-12-14 09:36:12 2016-12-14 16:36:12    ANDROID2-POI        6862205   
11 2016-12-14 09:40:18 2016-12-14 16:40:18    ANDROID2-POI       27596153   
12 2016-12-14 10:11:25 2016-12-14 17:11:25    ANDROID2-POI         294777   
13 2016-12-14 10:48:11 2016-12-14 17:48:11     IPHONE2-POI         588955   
14 2016-12-14 11:39:26 2016-12-14 18:39:26     IPHONE2-POI        3086705   
15 2016-12-14 12:49:06 2016-12-14 19:49:06    ANDROID2-POI       15673544   
16 2016-12-14 13:10:27 2016-12-14 20:10:27     IPHONE2-POI       27614281   
17 2016-12-15 00:20:13 2016-12-15 07:20:13     IPHONE2-POI        2032636   
18 2016-12-15 01:56:16 2016-12-15 08:56:16    ANDROID2-POI       25006211   
19 2016-12-15 07:13:03 2016-12-15 14:13:03    ANDROID2-POI        2039664   
20 2016-12-15 07:59:40 2016-12-15 14:59:40     IPHONE2-POI       20023619   
21 2016-12-15 09:02:56 2016-12-15 16:02:56    ANDROID2-POI       24216287   
22 2016-12-15 09:40:25 2016-12-15 16:40:25    ANDROID2-POI        3191748   
23 2016-12-15 09:55:16 2016-12-15 16:55:16    ANDROID2-POI        3191748   
24 2016-12-15 09:55:35 2016-12-15 16:55:35    ANDROID2-POI       27687033   
25 2016-12-15 10:03:50 2016-12-15 17:03:50    ANDROID2-POI        3396888   
26 2016-12-15 10:37:01 2016-12-15 17:37:01    ANDROID2-POI        6862205   
27 2016-12-15 13:01:18 2016-12-15 20:01:18    ANDROID2-POI       24216287   
28 2016-12-15 13:51:20 2016-12-15 20:51:20     IPHONE2-POI       10569690   
29 2016-12-16 02:09:36 2016-12-16 09:09:36    ANDROID2-POI       27687033   
..                 ...                 ...             ...            ...   
54 2016-12-16 11:05:08 2016-12-16 18:05:08    ANDROID2-POI       27791026   
55 2016-12-16 12:37:25 2016-12-16 19:37:25    ANDROID2-POI        1150271   
56 2016-12-16 12:59:02 2016-12-16 19:59:02     IPHONE2-POI        9097388   
57 2016-12-16 14:28:03 2016-12-16 21:28:03    ANDROID2-POI        4840240   
58 2016-12-16 16:44:28 2016-12-16 23:44:28    ANDROID2-POI        3810686   
59 2016-12-16 17:14:43 2016-12-17 00:14:43     IPHONE2-POI         503262   
60 2016-12-17 03:30:58 2016-12-17 10:30:58     IPHONE2-POI        3991235   
61 2016-12-17 03:53:32 2016-12-17 10:53:32    ANDROID2-POI        4748886   
62 2016-12-17 05:11:23 2016-12-17 12:11:23    ANDROID2-POI       18830579   
63 2016-12-17 06:07:11 2016-12-17 13:07:11    ANDROID2-POI        9268148   
64 2016-12-17 07:53:29 2016-12-17 14:53:29     IPHONE2-POI        4098204   
65 2016-12-17 08:12:33 2016-12-17 15:12:33    ANDROID2-POI       10592213   
66 2016-12-17 08:35:19 2016-12-17 15:35:19    ANDROID2-POI       24216287   
67 2016-12-17 08:53:01 2016-12-17 15:53:01     IPHONE2-POI        9423542   
68 2016-12-17 10:20:08 2016-12-17 17:20:08    ANDROID2-POI       11012398   
69 2016-12-17 12:33:03 2016-12-17 19:33:03    ANDROID2-POI       15444200   
70 2016-12-17 14:26:34 2016-12-17 21:26:34    ANDROID2-POI       27909533   
71 2016-12-17 17:55:03 2016-12-18 00:55:03     IPHONE2-POI       11592036   
72 2016-12-17 18:03:11 2016-12-18 01:03:11     IPHONE2-POI        4311153   
73 2016-12-17 18:47:32 2016-12-18 01:47:32     IPHONE2-POI       11592036   
74 20

In [36]:
seven_file.to_csv('seven_file.csv',sep=',',encoding='utf-8')

	creation_time	TH_time	distance_source	passenger1_id	passenger1_name	remarks	state	link
0	12/14/2016 6:24	12/14/2016 13:24	IPHONE2-POI	7066160	Wilaiwan Dhrammajariyapong	เธเธฃเธธเธ“เธฒเนเธชเธ”เธ‡เธ‚เน‰เธญเธกเธนเธฅเนƒเธ™เธฅเธดเธ‡เธ„เนŒเธ™เธตเน‰เนƒเธซเน‰เธžเธ™เธฑเธเธ‡เธฒเธ™ 7-Eleven. https://goo.gl/DDfl6O	COMPLETED	Oujdtytzbp
